### PR TITLE
STM Zero Suppression

### PR DESCRIPTION
**Pull request by AndrewEdmonds11**
Hi Everyone,

In this PR I:
 - add a new STM module (```STMReco/src/STMZeroSuppression_module.cc```) to perform the zero-suppression algorithm on the unsuppressed ```STMWaveformDigis```
    - note that the output data product type is the same as the input data product type
 - add a fcl file to run the zero suppression
    - note that I am defining the ```samplingFrequency``` parameter in fcl. This is used to convert from clock ticks to nanoseconds and, in the test beam data, the HPGe and LaBr data have different sampling frequencies. In the real experiment, they will have the same sampling frequency and I guess will be defined in a database somewhere
 - update ```PlotSTMWaveformDigis``` module and its fcl file to take advantage of the clock tick to nanosecond conversion so that we can plot zero-suppressed waveforms on top of the unsuppressed waveform
     - I also added an example macro that can be used to make such a plot

Once this is merged in, I will work on the Moving Window Deconvolution (MWD) algorithm, which will require a new data product.

Let me know if you have any questions or comments

Thanks,
Andy

### CODE REVIEW

**Comment by kutschke** [comment]
@FNALbuild run build teset

**Review by kutschke — COMMENTED** [review]
A few comments for now - more when I have a chance.

Is there anyone who can meaningfully review this from the STM point of view?

File: STMReco/fcl/prolog.fcl
Comment (rlcee) [reply STMReco/fcl/prolog.fcl:12]: Andy checked in with me before making this commit.  I feel it is part of my job to push back against making too many tables or making them too early because I don't want to take maintenance and efficiency lightly, since it is hard to measure and will sneak up on us.  We can talk more.  Once the decision is made, I'm ready to support.

**Comment by kutschke** [comment]
@FNALbuild run build test

File: STMReco/src/PlotSTMWaveformDigis_module.cc
Comment (AndrewEdmonds11) [reply STMReco/src/PlotSTMWaveformDigis_module.cc:86]: Thanks, Rob. I've done that in commit [b8eeea8](https://github.com/Mu2e/Offline/pull/917/commits/b8eeea8644ceaf399e0a943dbb9b14d942a0e9e6). Is something like that what you had in mind?

Comment (kutschke) [reply STMReco/src/PlotSTMWaveformDigis_module.cc:86]: Yes.  That's what I had in mind.   In the branch for "waveform_time" is there a typo: xmin should be tmin?

And a followup question.  Is it safe to use just the event number as the histogram identifier?  Will you ever run this code on a set of files  with multiple runs/subruns so that event numbers might be duplicated?

Comment (AndrewEdmonds11) [reply STMReco/src/PlotSTMWaveformDigis_module.cc:86]: No that's not a typo. I want to define "waveform_time" as starting at 0. I willchange the variable names to make it clearer.

I have mixed feelings about adding the run and subrun number to the histogram name. I don't think we would run this code over multiple runs at once - it is supposed to be just a debugging tool - and it is easier to run a plotting macro on different files if we don't always need to specify a run and subrun number. However, we may want to compare waveforms between runs... I think I'd like to leave it as-is for now and revisit if we find we've made the wrong choice

Comment (kutschke) [reply STMReco/src/PlotSTMWaveformDigis_module.cc:86]: You might add a comment to notify future STM maintainers about that time convention.  If it's written up elsewhere, just reference it.

About the histogram identifier.  Your call.   Either way I wanted it to be a considered decision.

Comment (kutschke) [reply STMReco/src/PlotSTMWaveformDigis_module.cc:86]: ... if you can tell the story well  enough with wisely chosen variable names, that better than a comment.

Comment (AndrewEdmonds11) [reply STMReco/src/PlotSTMWaveformDigis_module.cc:86]: I've chosen (hopefully) better variable names in [c1f1e6d](https://github.com/Mu2e/Offline/pull/917/commits/c1f1e6d35fb3b1442d1f8c63b6e92434ad5ed1e2)

**Comment by kutschke** [comment]
Have the discussions with Ray on this PR converged?

**Comment by kutschke** [comment]
@FNALbuild run build test

**Review by kutschke — APPROVED** [review]
One suggestion for thinking about the future.  If you think that you may want to use zero supression code in more than one module, put that code into it's own classes/functions and make the module a thin wrapper that interfaces the art environment to the algorithm.  

Ray and Andy tell me that their discussion about appropriate locations for the frequency settings has converged.  It's in the right place for now.

**Pull request merged**
